rocketpy/Function.py

```diff
@@ -2024,9 +2024,262 @@ def integral(self, a, b, numerical=False):
+            value is the end of the dataset.
+        datapoints : int, optional
+            The number of points in which the integral will be evaluated for
+            plotting it, which draws lines between each evaluated point.
+            The default value is 100.
+
+        Returns
+        -------
+        result : Function
+            The integral of the Function object.
+        """
+        # Check if lower and upper are given
+        if lower is None:
+            if isinstance(self.source, np.ndarray):
+                lower = self.source[0, 0]
+            else:
+                raise ValueError("Lower limit must be given if source is a function.")
+        if upper is None:
+            if isinstance(self.source, np.ndarray):
+                upper = self.source[-1, 0]
+            else:
+                raise ValueError("Upper limit must be given if source is a function.")
+
+        # Create a new Function object
+        xData = np.linspace(lower, upper, datapoints)
+        yData = np.zeros(datapoints)
+        for i in range(datapoints):
+            yData[i] = self.integral(lower, xData[i])
+        return Function(np.concatenate(([xData], [yData])).transpose(), 
+                    inputs=self.__inputs__, 
+                    outputs=[o + " Integral" for o in self.__outputs__])
+
+    def invert(self, lower=None, upper=None, datapoints=100):
+        """
+        Inverts the Function. Makes the domain the range and the range the domain.
+
+        Parameters
+        ----------
+        lower : float
+            Lower limit of the new domain.
+        upper : float
+            Upper limit of the new domain.
+
+        Returns
+        -------
+        result : Function
+            A Function whose domain and range have been inverted.
+        """
+        # Check if lower and upper are given
+        if lower is None:
+            if isinstance(self.source, np.ndarray):
+                lower = self.source[0, 0]
+            else:
+                raise ValueError("Lower limit must be given if source is a function.")
+        if upper is None:
+            if isinstance(self.source, np.ndarray):
+                upper = self.source[-1, 0]
+            else:
+                raise ValueError("Upper limit must be given if source is a function.")
+
+        # Create a new Function object
+        xData = np.linspace(lower, upper, datapoints)
+        yData = np.zeros(datapoints)
+        for i in range(datapoints):
+            yData[i] = self.getValue(xData[i])
+
+        return Function(np.concatenate(([yData], [xData])).transpose(),
+                    inputs=self.__outputs__,
+                    outputs=self.__inputs__)
+
+    def findOptimalInput(self, val):
+        """
+        Finds the optimal input for a given output.
+
+        Parameters
+        ----------
+        val : float
+            The value of the output.
+
+        Returns
+        -------
+        result : float
```

Comment:
The result of [`scipy.optimize.fmin`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.fmin.html) is a `ndarray` (even if it has only one element). This does not cause problems since `ndarray` supports most mathematical operations that a `float` does.

```suggestion
        result : ndarray
```